src/smart-components/role/add-role-permissions/add-role-permission-view.js

```diff
@@ -0,0 +1,98 @@
+import React, { useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { shallowEqual, useSelector, useDispatch, batch } from 'react-redux';
+import { listPermissions, listPermissionOptions } from '../../../redux/actions/permission-action';
+import { TableToolbarView } from '../../../presentational-components/shared/table-toolbar-view';
+
+const columns = ['Application', 'Resource type', 'Operation'];
+const selector = ({
```

Comment:
Just a note for future refactoring. This selector is returning non-primitive object values so the `shallowEqual` won't prevent all unnecessary triggers which will lead to re-renders.

The selector should be split into multiple selectors using multiple `useSelector` hooks returning each value separatetly to improve the performance.